### PR TITLE
fix: allow any prop for some specific $w.Element usages 

### PR DESCRIPTION
**Pull request by dor-itzhaki**
This PR adds 3 AST transformation to our main declarations.d.ts file.
The transformation will allow using any property in 3 cases that are currently defined as $w.Element due to lack of "generics" support in our documentation.
Note that the user will still get autocomplete for the $w.Element members/methods, but will not get errors when using unknown properties/methods (which hopefully belong to specific elements).

I didn't add any tests yet + due to the upgrade of typescript, some snapshots are broken.
I'll fix all of that later, but i'd like to get your feedback.

### CODE REVIEW

File: scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  const playableMixin = $w.body.statements.find(
+    statement =>
+      ts.isInterfaceDeclaration(statement) &&
+      statement.name.escapedText === "PlayableMixin"
+  );
+  const playableMixinNext = playableMixin.members.find(
+    member => member.name.escapedText === "next"
+  );
+
+  playableMixinNext.type.typeArguments[0] = ts.factory.createIntersectionTypeNode(
```
Comment (nemni8) [inline scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js:17]: what does [0] represent?

File: scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+function allowAnyPropOnPlayableMixinNext(ast) {
+  const $w = ast.statements.find(
+    statement =>
+      ts.isModuleDeclaration(statement) && statement.name.escapedText === "$w"
+  );
+  const playableMixin = $w.body.statements.find(
+    statement =>
+      ts.isInterfaceDeclaration(statement) &&
+      statement.name.escapedText === "PlayableMixin"
```
Comment (nemni8) [inline scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js:11]: I would extract all hardcoded string to one consts file

File: scripts/ast-patches/allowAnyPropOnContainerChildren.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const ts = require("typescript");
+
+function allowAnyPropOnContainerChildren(ast) {
+  const $w = ast.statements.find(
```
Comment (nemni8) [inline scripts/ast-patches/allowAnyPropOnContainerChildren.js:4]: must of the code in the ast-patches are duplicated with minor changes. 
I would extract the login to one place

File: scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  );
+  const playableMixinNext = playableMixin.members.find(
+    member => member.name.escapedText === "next"
```
Comment (nemni8) [inline scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js:14]: you probably should handle `previous` members as well

**Review by nemni8 — COMMENTED** [review]
great work! 
I think that we should also handle (maybe not as part of this task) in all other cases that return $w.Element. for example:
`onInput(eventHandler: $w.EventHandler): $w.Element;`
`onViewportEnter(handler: $w.EventHandler): $w.Element;` 
or any other $w mixin events defenition

File: scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js (hunk 1 above)
Comment (dor-itzhaki) [reply scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js:17]: that's the way a promise is constructed in ast

File: scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js (hunk 2 above)
Comment (dor-itzhaki) [reply scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js:11]: i don't think it'll give any value here... it's only used once

File: scripts/ast-patches/allowAnyPropOnContainerChildren.js (hunk 3 above)
Comment (dor-itzhaki) [reply scripts/ast-patches/allowAnyPropOnContainerChildren.js:4]: i'll extract a few common queries...

File: scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js (hunk 4 above)
Comment (dor-itzhaki) [reply scripts/ast-patches/allowAnyPropOnPlayableMixinNext.js:14]: will do, thanks

File: scripts/ast-patches/allowAnyPropOnContainerChildren.js (hunk 3 above)
Comment (dor-itzhaki) [reply scripts/ast-patches/allowAnyPropOnContainerChildren.js:4]: actually, it seems like this creates more mess imo, since every case is a bit different

**Comment by dor-itzhaki** [comment]
> great work!
> I think that we should also handle (maybe not as part of this task) in all other cases that return $w.Element. for example:
> `onInput(eventHandler: $w.EventHandler): $w.Element;`
> `onViewportEnter(handler: $w.EventHandler): $w.Element;`
> or any other $w mixin events defenition

You're right but i think those are less popular...
I would actually like to try and create an AST transformation that will make our mixins generic, but holding it off for now

**Pull request merged**